flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -166,7 +169,8 @@ class JobManager(
       taskManager,
       connectionInfo,
       hardwareInformation,
-      numberOfSlots) =>
+      numberOfSlots,
+      taskManagerVersionID) =>
```

Comment:
Why do you check the version when a `TaskManager` registers at the `JobManager`. I thought the critical part would be the `JobClient` and `JobManager` communication.